src/dbos-runtime/config.ts

```diff
@@ -120,23 +120,31 @@ export function constructPoolConfig(configFile: ConfigFile, cliOptions?: ParseOp
   const databaseConnection = loadDatabaseConnection();
   if (!cliOptions?.skipLoggingInParse) {
     const logger = new GlobalLogger();
-    if (configFile['database']['hostname']) {
+    if (process.env.DBOS_DBHOST) {
+      logger.info('Loading database connection parameters from debug environment variables');
+    } else if (configFile.database.hostname) {
       logger.info('Loading database connection parameters from dbos-config.yaml');
-    } else if (databaseConnection['hostname']) {
+    } else if (databaseConnection.hostname) {
       logger.info('Loading database connection parameters from .dbos/db_connection');
     } else {
       logger.info('Using default database connection parameters');
     }
   }
-  configFile['database']['hostname'] =
-    configFile['database']['hostname'] || databaseConnection['hostname'] || 'localhost';
-  configFile['database']['port'] = configFile['database']['port'] || databaseConnection['port'] || 5432;
-  configFile['database']['username'] =
-    configFile['database']['username'] || databaseConnection['username'] || 'postgres';
-  configFile['database']['password'] =
-    configFile['database']['password'] || databaseConnection['password'] || process.env.PGPASSWORD || 'dbos';
-  configFile['database']['local_suffix'] =
-    configFile['database']['local_suffix'] || databaseConnection['local_suffix'] || false;
+  configFile.database.hostname =
+    process.env.DBOS_DBHOST ?? configFile.database.hostname ?? databaseConnection.hostname ?? 'localhost';
```

Comment:
Why did we change to `??` (nullish coalescing) instead of `||` (the OR operator)? I think the benefit of `||` is that if the env variable is set to an empty string, it will consider to be not set. Otherwise, we might get some empty string variables.